libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -76,9 +76,15 @@ AP_BattMonitor::init()
                 _num_instances++;
                 break;
             case AP_BattMonitor_Params::BattMonitor_TYPE_MAXELL:
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_PX4_V3
```

Comment:
This is wrong, PX4-v3 isn't just Pixhawk2. Also, Pixhawk2 has the two I2C buses on the carrier board so we can't just assume it's connected on the I2C2 port. What we need is a probe method for these I2C monitors, so that it can be detected if they are there.